src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -40,6 +40,7 @@ public ShoppingCart getByUser(User user) {
     public void registerNewShoppingCart(User user) {
         ShoppingCart shoppingCart = new ShoppingCart();
         shoppingCart.setUser(user);
+        shoppingCart.setId(user.getId());
```

Comment:
isn't it set automatically with @MapsId annotation?

Reply:
Sorry forgot to clear